library/src/com/google/maps/android/geojson/GeoJsonFeature.java

```diff
@@ -0,0 +1,211 @@
+    }
+
+    /**
+     * Sets the GeoJsonLineStringStyle of the feature
+     *
+     * @param geoJsonLineStringStyle GeoJsonLineStringStyle object
+     */
+    public void setLineStringStyle(GeoJsonLineStringStyle geoJsonLineStringStyle) {
+        mGeoJsonLineStringStyle.deleteObserver(this);
+        mGeoJsonLineStringStyle = geoJsonLineStringStyle;
+        mGeoJsonLineStringStyle.addObserver(this);
+        checkRedrawFeature(mGeoJsonLineStringStyle);
+    }
+
+    /**
+     * Gets the GeoJsonPolygonStyle of the feature
+     *
+     * @return GeoJsonPolygonStyle object
+     */
+    public GeoJsonPolygonStyle getPolygonStyle() {
+        return mGeoJsonPolygonStyle;
+    }
+
+    /**
+     * Sets the GeoJsonPolygonStyle of the feature
+     *
+     * @param geoJsonPolygonStyle GeoJsonPolygonStyle object
+     */
+    public void setPolygonStyle(GeoJsonPolygonStyle geoJsonPolygonStyle) {
+        mGeoJsonPolygonStyle.deleteObserver(this);
+        mGeoJsonPolygonStyle = geoJsonPolygonStyle;
+        mGeoJsonPolygonStyle.addObserver(this);
+        checkRedrawFeature(mGeoJsonPolygonStyle);
+    }
+
+    /**
+     * Checks whether the new style that was set requires the feature to be redrawn. If the
+     * geometry
+     * and the style that was set match, then the feature is redrawn.
+     *
+     * @param style style to check if a redraw is needed
+     */
+    private void checkRedrawFeature(GeoJsonStyle style) {
+        if (mGeoJsonGeometry != null && mGeoJsonGeometry.getType()
+                .matches(style.getGeometryType())) {
+            setChanged();
+            notifyObservers();
+        }
+    }
+
+    /**
+     * Gets the stored GeoJsonGeometry object
+     *
+     * @return GeoJsonGeometry object
+     */
+    public GeoJsonGeometry getGeometry() {
+        return mGeoJsonGeometry;
+    }
+
+    /**
+     * Sets the stored GeoJsonGeometry object and redraws it to the map if it has been added to the
+     * layer
+     *
+     * @param geometry GeoJsonGeometry object to set
+     */
+    public void setGeometry(GeoJsonGeometry geometry) {
+        mGeoJsonGeometry = geometry;
+        setChanged();
+        notifyObservers();
+    }
+
+    /**
+     * Gets the ID of the feature
+     *
+     * @return ID of the feature
+     */
+    public String getId() {
+        return mId;
+    }
+
+    /**
+     * Checks if the geometry is assigned
+     *
+     * @return true if feature contains geometry object, false if geometry is currently null
+     */
+    public boolean hasGeometry() {
+        return (mGeoJsonGeometry != null);
+    }
+
+    /**
+     * Gets the array containing the coordinates of the bounding box for the FeatureCollection. If
+     * the FeatureCollection did not have a bounding box or if the GeoJSON file did not contain a
+     * FeatureCollection then null will be returned.
+     *
+     * @return LatLngBounds containing bounding box of FeatureCollection, null if no bounding box
```

Comment:
FeatureCollection is correct. FeatureCollections can have a bounding box but so can features as well. We're representing the layer as the feature collection as we don't have a class specifically for that.